forum/models/comments.py

```diff
@@ -0,0 +1,149 @@
+        votes: Optional[Dict[str, int]] = None,
+        abuse_flaggers: Optional[List[str]] = None,
+        endorsed: Optional[bool] = None,
+        child_count: Optional[int] = None,
+        depth: Optional[int] = None,
+    ) -> int:
+        """
+        Updates a comment document in the database.
+
+        Args:
+            comment_id (ObjectId): The ID of the comment to update.
+            body (Optional[str], optional): The body content of the comment.
+            course_id (Optional[str], optional): The ID of the course the comment is associated with.
+            anonymous (Optional[bool], optional): Whether the comment is posted anonymously.
+            anonymous_to_peers (Optional[bool], optional): Whether the comment is anonymous to peers.
+            comment_thread_id (Optional[ObjectId], optional): The ID of the parent comment thread.
+            at_position_list (Optional[List[str]], optional): A list of positions for @mentions.
+            visible (Optional[bool], optional): Whether the comment is visible.
+            author_id (Optional[str], optional): The ID of the author who created the comment.
+            author_username (Optional[str], optional): The username of the author.
+            votes (Optional[Dict[str, int]], optional): The votes for the comment.
+            abuse_flaggers (Optional[List[str]], optional): A list of users who flagged the comment for abuse.
+            endorsed (Optional[bool], optional): Whether the comment is endorsed.
+            child_count (Optional[int], optional): The number of child comments.
+            depth (Optional[int], optional): The depth of the comment in the thread hierarchy.
+
+        Returns:
+            int: The number of documents modified.
+        """
+        update_data: Dict[str, Any] = {}
+        if body:
+            update_data["body"] = body
+        if course_id:
+            update_data["course_id"] = course_id
+        if anonymous is not None:
+            update_data["anonymous"] = anonymous
+        if anonymous_to_peers is not None:
+            update_data["anonymous_to_peers"] = anonymous_to_peers
+        if comment_thread_id:
+            update_data["comment_thread_id"] = comment_thread_id
+        if at_position_list:
+            update_data["at_position_list"] = at_position_list
+        if visible is not None:
+            update_data["visible"] = visible
+        if author_id:
+            update_data["author_id"] = author_id
+        if author_username:
+            update_data["author_username"] = author_username
+        if votes:
+            update_data["votes"] = votes
+        if abuse_flaggers:
+            update_data["abuse_flaggers"] = abuse_flaggers
+        if endorsed is not None:
+            update_data["endorsed"] = endorsed
+        if child_count is not None:
+            update_data["child_count"] = child_count
+        if depth is not None:
+            update_data["depth"] = depth
```

Comment:
we can reduce the if-else code as done in (users and threads)'s update method.
like below
```
fields = [
    ("body", body),
    ("course_id", course_id),
    ("anonymous", anonymous),
    ("anonymous_to_peers", anonymous_to_peers),
    ("comment_thread_id", comment_thread_id),
    ("at_position_list", at_position_list),
    ("visible", visible),
    ("author_id", author_id),
    ("author_username", author_username),
    ("votes", votes),
    ("abuse_flaggers", abuse_flaggers),
    ("endorsed", endorsed),
    ("child_count", child_count),
    ("depth", depth),
]

update_data: Dict[str, Any] = {field: value for field, value in fields if value is not None}
```